packages/@guardian/source-foundations/src/typography/data.ts

```diff
@@ -1,243 +1,153 @@
-		},
-		bold: {
-			hasItalic: false,
-		},
+export const underlineThickness = {
+	textSans: {
+		xxsmall: 2,
+		xsmall: 2,
+		small: 2,
+		medium: 2,
+		large: 3,
+		xlarge: 3,
+		xxlarge: 3,
+		xxxlarge: 4,
 	},
 	body: {
-		regular: {
-			hasItalic: true,
-		},
-		bold: {
-			hasItalic: true,
-		},
+		xsmall: 2,
+		small: 2,
+		medium: 2,
 	},
-	textSans: {
-		regular: {
-			hasItalic: true,
-		},
-		bold: {
-			hasItalic: false,
-		},
+	headline: {
+		xxxsmall: 2,
+		xxsmall: 3,
+		xsmall: 3,
+		small: 3,
+		medium: 4,
+		large: 5,
+		xlarge: 6,
+	},
+	titlepiece: {
+		small: 5,
+		medium: 6,
+		large: 6,
 	},
-};
-
-const titlepieceUnderlineThickness = {
-	small: underlineThickness[3], //5px
-	medium: underlineThickness[4], //6px
-	large: underlineThickness[4], //6px
 } as const;
 
-const headlineUnderlineThickness = {
-	xxxsmall: underlineThickness[0], //2px
-	xxsmall: underlineThickness[1], //3px
-	xsmall: underlineThickness[1], //3px
-	small: underlineThickness[1], //3px
-	medium: underlineThickness[2], //4px
-	large: underlineThickness[3], //5px
-	xlarge: underlineThickness[4], //6px
-} as const;
+// Pixel font size exports
+export const textSansSizes = pxTextSizes.textSansSizes;
 
-const bodyUnderlineThickness = {
-	xsmall: underlineThickness[0], //2px
-	small: underlineThickness[0], //2px
-	medium: underlineThickness[0], //2px
-} as const;
+export const bodySizes = pxTextSizes.bodySizes;
 
-const textSansUnderlineThickness = {
-	xxsmall: underlineThickness[0], //2px
-	xsmall: underlineThickness[0], //2px
-	small: underlineThickness[0], //2px
-	medium: underlineThickness[0], //2px
-	large: underlineThickness[1], //3px
-	xlarge: underlineThickness[1], //3px
-	xxlarge: underlineThickness[1], //3px
-	xxxlarge: underlineThickness[2], //4px
-} as const;
+export const headlineSizes = pxTextSizes.headlineSizes;
+
+export const titlepieceSizes = pxTextSizes.titlepieceSizes;
+
+// Computed rem font size exports
+
+export const remTitlepieceSizes = remTextSizes.titlepieceSizes;
 
-const underlineThicknessMapping = {
-	titlepiece: titlepieceUnderlineThickness,
-	headline: headlineUnderlineThickness,
-	body: bodyUnderlineThickness,
-	textSans: textSansUnderlineThickness,
-};
+export const remHeadlineSizes = remTextSizes.headlineSizes;
 
-Object.freeze(titlepieceSizes);
-Object.freeze(headlineSizes);
-Object.freeze(bodySizes);
-Object.freeze(textSansSizes);
-Object.freeze(remTitlepieceSizes);
-Object.freeze(remHeadlineSizes);
-Object.freeze(remBodySizes);
-Object.freeze(remTextSansSizes);
-Object.freeze(fontMapping);
-Object.freeze(fontSizeMapping);
-Object.freeze(fontWeightMapping);
-Object.freeze(lineHeightMapping);
-Object.freeze(availableFonts);
-Object.freeze(titlepieceUnderlineThickness);
-Object.freeze(headlineUnderlineThickness);
-Object.freeze(bodyUnderlineThickness);
-Object.freeze(textSansUnderlineThickness);
-Object.freeze(underlineThicknessMapping);
+export const remBodySizes = remTextSizes.bodySizes;
 
-export {
-	titlepieceSizes,
```

Comment:
Is there a case here that listing all exports gives an overview of what's within the file? Not a strong opinion..

Reply:
I agree - having the overview down below is really useful.

I changed it to a per-line export because with that system you can have an overview of the references to each object.. which is massively helpful when doing stuff like refactors, or any modification in general.


Here's a demo of the difference:
<img width="396" alt="Screenshot 2022-09-05 at 14 44 59" src="https://user-images.githubusercontent.com/1771189/188463912-69729b5a-8236-4325-9eeb-d82f1d1288c1.png">

<img width="675" alt="Screenshot 2022-09-05 at 14 44 22" src="https://user-images.githubusercontent.com/1771189/188463924-20f94dc9-9d06-4ea9-abed-320a10f439f1.png">

Reply:
Yeah that nice to have visibility over references - I guess with this one again maybe consistency could be our guide.. if we use grouped exports elsewhere maybe its good to do that here? 

Happy with whatever the consensus is thouugh.

Reply:
agreed — there might be a way to have grouped exports and retain the references, will do a little digging